metrics-exporter-prometheus/src/exporter/push_gateway.rs

```diff
@@ -19,7 +18,12 @@ pub(super) fn new_push_gateway(
     handle: PrometheusHandle,
 ) -> ExporterFuture {
     Box::pin(async move {
-        let https = HttpsConnector::new();
+        let https = hyper_rustls::HttpsConnectorBuilder::new()
+            .with_native_roots()
+            .expect("no native root CA certificates found")
+            .https_only()
```

Comment:
This change will only allow pushing to HTTPS endpoints, whereas the old code allows for either HTTP or HTTPS.

This should instead of `https_or_http`.

Reply:
yes, of course. sorry about that.

Reply:
done